src/main/java/net/neoforged/neoforge/common/conditions/ElementExistsCondition.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.conditions;
+
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.Holder;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+
+public record ElementExistsCondition<TRegistry>(ResourceKey<TRegistry> registryKey) implements ICondition {
+    public static final MapCodec<ElementExistsCondition<?>> CODEC = RecordCodecBuilder.mapCodec(instance -> instance
+            .group(ResourceLocation.CODEC.lenientOptionalFieldOf("registry", Registries.ITEM.location()).forGetter(condition -> condition.registryKey().registry()),
```

Comment:
Don't use lenient optional, it might swallow errors. I wonder if `ResourceKey.codec(Registries.ROOT_REGISTRY_NAME)` could be used here.

Reply:
`ResourceKey.codec` wants a `ResourceKey<? extends Registry<T>>`, `ResourceKey.ROOT_REGISTRY_NAME` is a `ResourceLocation`.
Dont think i can use that method due to it being typed and the registries resource key.

The way i have also designed the codec here means existing json files outside of renaming the `item` value to `element` should just work.

Reply:
Yes it's good then, just replace the lenient optional by a normal optional. Lenient optional will cause nonsensical values such as `{}` to be silently ignored.